spec/lib/dor/text_extraction/speech_to_text_spec.rb

```diff
@@ -105,28 +105,41 @@
   describe '#language_tag' do
     context 'when the file cannot be found' do
       let(:filename) { 'bogus.mp4' }
+      let(:s3_filename) { 'bogus_mp4.mp4' }
 
       it 'returns nil' do
         expect(stt.filenames_to_stt).not_to include(filename)
-        expect(stt.language_tag(filename)).to be_nil
+        expect(stt.language_tag(s3_filename)).to be_nil
       end
     end
 
     context 'when the file is found and there is no language tag in cocina' do
       let(:filename) { 'file1.m4a' }
+      let(:s3_filename) { 'file1_m4a.m4a' }
 
       it 'returns nil' do
         expect(stt.filenames_to_stt).to include(filename)
-        expect(stt.language_tag(filename)).to be_nil
+        expect(stt.language_tag(s3_filename)).to be_nil
       end
     end
 
     context 'when the file is found and there is a language tag in cocina' do
       let(:filename) { 'file1.mp4' }
+      let(:s3_filename) { 'file1_mp4.mp4' }
 
       it 'returns the language tag' do
         expect(stt.filenames_to_stt).to include(filename)
-        expect(stt.language_tag(filename)).to eq 'es'
+        expect(stt.language_tag(s3_filename)).to eq 'es'
+      end
+    end
+
+    context 'when the file is found, the s3 filename is the same as the cocina filename and there is a language tag in cocina' do
+      let(:filename) { 'file1.mp4' }
+      let(:s3_filename) { 'file1.mp4' }
+
+      it 'returns the language tag' do
+        expect(stt.filenames_to_stt).to include(filename)
+        expect(stt.language_tag(s3_filename)).to eq 'es'
```

Comment:
our new method will also work even if the filenames match exactly between s3 and cocina